### PR TITLE
Add refined ML-SUPERB2 MMS CTC recipes

### PR DESCRIPTION
**Pull request by Shunsuke32**
## What did you change?

Updated the ML-SUPERB 2.0 ASR recipe with LID-label normalization in the default data preparation.

The existing `local/download.py` now applies the following LID-label handling:

- `lga -> lug`
- `ory -> ori`
- `arb -> ara`
- removes utterances labeled as `nno`, `nob`, or `nor`

The separate refined data-preparation files were removed, and the default `local/data.sh` path is kept.

Updated the default ASR config:

- `conf/train_asr.yaml` now points to the E-Branchformer configuration.

Added MMS 1B + CTC training configs:

- 24-layer Transformer encoder
- Conformer encoder
- E-Branchformer encoder

The original 2-layer Transformer baseline config is kept as `conf/tuning/train_mms_baseline.yaml`.

The Transformer, Conformer, and E-Branchformer downstream models are configured to keep the number of trainable parameters below 100M.

Updated the ML-SUPERB 2.0 README with:

- available configs
- challenge-style results
- Hugging Face model links

## Why did you make this change?

The ML-SUPERB 2.0 public data contains a few LID labels that should be normalized or removed according to the challenge data description.

This PR applies those corrections in the default data-preparation path instead of maintaining a separate refined path.

This PR also updates the default model to E-Branchformer based on the reported results, while keeping the original 2-layer Transformer baseline available as a tuning config.

I also tried `keep_nbest_models: 4`, but it did not clearly improve the results and decreased Dialect LID by about 1-4 points, so I kept `keep_nbest_models: 2`.

## Is your PR small enough?

6 files changed, with 330 additions and 44 deletions.

The diff is limited to the ML-SUPERB 2.0 ASR recipe:

- 1 README update
- 1 data-preparation script update
- 1 default ASR config symlink update
- 3 training configs

### CODE REVIEW

**Comment by sw005320** [comment]
Can you paste the learning curves?
I just want to check what they look like for each condition.

**Comment by Shunsuke32** [comment]
Yes, of course. I have attached the screenshots of the training and validation loss curves below.
image
image

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+batch_type: sorted
+batch_size: 8
+accum_grad: 4
+patience: none
+init: none
+best_model_criterion:
+-   - valid
+    - loss
+    - min
+keep_nbest_models: 2
```
Comment (sw005320) [inline egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:52]: where does this come from?
you may increase this.

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+encoder: conformer
+encoder_conf:
+    output_size: 512
+    attention_heads: 8
+    linear_units: 2048
+    num_blocks: 12
+    dropout_rate: 0.1
+    positional_dropout_rate: 0.1
+    attention_dropout_rate: 0.1
+    input_layer: conv2d2
```
Comment (sw005320) [inline egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:10]: Can you also try `conv2d`?
As I understand it, `mms` already performs downsampling.
I'm concerned that too much downsampling may hurt the performance (but it would require more computing).

**Comment by sw005320** [comment]
> Yes, of course. I have attached the screenshots of the training and validation loss curves below. image image

thx!
Can you combine train and valid in the graph? espnet actually generates a PNG file.

**Comment by sw005320** [comment]
- They look like they are not fully converged
- The training loss is too large (but possibly due to the specaug and dropout). Can you also share the training/validation accuracies?

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml (hunk 2 above)
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:10]: Sure, I will try `conv2d`. I was originally following the baseline configuration (a 2-layer Transformer) provided in this repository, but I will investigate the downsampling behavior on my end as well.

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml (hunk 1 above)
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:52]: This is the default value from the repository's baseline. I agree that averaging more checkpoints (e.g., 4) would help stabilize the performance, so I'll try to increase this.

**Comment by Shunsuke32** [comment]
> They look like they are not fully converged
> The training loss is too large (but possibly due to the specaug and dropout). Can you also share the training/validation accuracies?

I agree that they are not fully converged. I will increase the total training steps to train the models longer.

Regarding the metrics, accuracies are not logged since these runs are CTC-only (`ctc_weight: 1.0`). Additionally, ESPnet does not log training CER, making a direct train/valid comparison impossible.

> thx!
> Can you combine train and valid in the graph? espnet actually generates a PNG file.

Thank you for the tip! I have attached the combined graphs generated by ESPnet below. 
From top to bottom, they correspond to the Transformer, Conformer, and E-Branchformer.

image
image
image

**Comment by Shunsuke32** [comment]
@sw005320 

I have updated this PR.

The main changes are:

- Added refined LID-label data preparation.
- Fixed the training schedule for the reported experiments. The previous setting made the total number of updates half of what was intended, so I doubled `num_iters_per_epoch` and reran the experiments.

I would appreciate any feedback on the updated PR when you have a chance.

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml (hunk 1 above)
Comment (sw005320) [reply egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:52]: I'm concerned about this (only 2 models) since it may not provide much benefit for model averaging.
Can you try the other numbers?

File: egs2/ml_superb2/asr1/local/data_refine.sh
Comment (sw005320) [inline egs2/ml_superb2/asr1/local/data_refine.sh:1]: You can replace it with local/data.sh
We don't need to maintain two files (also, this will confuse users)

File: egs2/ml_superb2/asr1/local/download_refined.py
Comment (sw005320) [inline egs2/ml_superb2/asr1/local/download_refined.py:1]: ditto
You can replace it with local/download.py
We don't need to maintain two files (also, this will confuse users)

File: egs2/ml_superb2/asr1/README.md [hunk 3]
```diff
@@ -74,3 +100,40 @@ Such that `references[i]`, `lids[i]`, and `hyps[i]` should all correspond to the
...
+- `accum_grad: 4`
+- `num_iters_per_epoch: 20000`
+- `max_epoch: 20`
+
+|model_name|decode_dir|Standard CER|Standard LID|Worst 15 CER|CER StD|Dialect CER|Dialect LID|
+|---|---|---:|---:|---:|---:|---:|---:|
+|Baseline 2-layer Transformer|decode_asr_asr_model_valid.loss.ave_2best|22.6|76.3|55.7|14.6|33.9|59.3|
+|Transformer 24-layer|decode_asr_asr_model_valid.loss.ave_2best|19.5|81.3|52.3|14.4|36.2|63.1|
+|Conformer|decode_asr_asr_model_valid.loss.ave_2best|21.3|72.5|57.9|16.0|39.6|62.0|
+|E-Branchformer|decode_asr_asr_model_valid.loss.ave_2best|18.6|81.8|51.1|14.4|33.5|72.3|
```
Comment (sw005320) [inline egs2/ml_superb2/asr1/README.md:105]: Given this great result, you can replace https://github.com/Shunsuke32/espnet/blob/5d6621511304c2bc00d8034334231e5ec5575196/egs2/ml_superb2/asr1/conf/train_asr.yaml with https://github.com/espnet/espnet/pull/6449/changes#diff-37067878f5b9110e2e7ea418681a4e84eab021cf4a5215eafb13cd52b81b3443

**Comment by Shunsuke32** [comment]
Thank you for the comments.

I will address them in the next update as follows:

- Replace the existing `local/data.sh` and `local/download.py` directly instead of adding separate refined versions.
- Keep the original README result table, but clarify that it was obtained before the LID-label normalization.
- Replace the default `conf/train_asr.yaml` with the E-Branchformer configuration.
- Rerun the experiments with `keep_nbest_models: 4`.

I will update this PR once the new experiments with `keep_nbest_models: 4` are finished.

File: egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml (hunk 1 above)
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/conf/tuning/train_mms_conformer.yaml:52]: Due to GPU and storage shortage on the current server, I am currently unable to finish the experiments quickly with a larger `keep_nbest_models` setting.
After I join AIST in August, I will rerun the experiments with increased `keep_nbest_models` and update the results accordingly.

File: egs2/ml_superb2/asr1/local/data_refine.sh
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/local/data_refine.sh:1]: Thank you. I will replace the existing `local/data.sh` instead.

File: egs2/ml_superb2/asr1/local/download_refined.py
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/local/download_refined.py:1]: Thank you. I will replace the existing local/downlaod.sh instead.

File: egs2/ml_superb2/asr1/README.md (hunk 3 above)
Comment (Shunsuke32) [reply egs2/ml_superb2/asr1/README.md:105]: Thank you for the suggestion. I will update the default `conf/train_asr.yaml` with the E-Branchformer configuration.

**Comment by Shunsuke32** [comment]
@sw005320
I updated the PR based on your comments.
The main updates are:

- Removed the separate `local/data_refine.sh` and `local/download_refined.py`.
- Merged the LID-label normalization into the existing `local/download.py`, so the default `local/data.sh` path is used.
- Updated `conf/train_asr.yaml` to use the E-Branchformer configuration.
- Tried `keep_nbest_models: 4`, but the difference was very small and Dialect LID decreased by about 1-4 points, so I kept `keep_nbest_models: 2`.

I would appreciate it if you could check the updated PR when you have time.

**Review by sw005320 — APPROVED** [review]
@Shunsuke32, any ongoing work?
If not, I'll merge this PR after the CI test.

**Comment by Shunsuke32** [comment]
Nothing else on my end. Please go ahead, thank you!

**Comment by sw005320** [comment]
Thanks, @Shunsuke32!
